Ginger/GingerCoreNET/ActionsLib/APIModelActionConversion/ApiActionConversionUtils.cs

```diff
@@ -0,0 +1,424 @@
+                            }
+                        }
+                        bf.ConversionStatus = eConversionStatus.Finish;
+                    }
+                });
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error occurred while trying to convert action", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method will create the new api model action
+        /// </summary>
+        /// <param name="applicationModel"></param>
+        /// <param name="act"></param>
+        /// <returns></returns>
+        private ActWebAPIModel GetNewAPIModelAction(Guid applicationModelGuid, Act act)
+        {
+            ActWebAPIModel actModel = new ActWebAPIModel();
+            actModel.ItemName = string.Format("New - {0}", act.ItemName);
+            actModel.Description = string.Format("New - {0}", act.Description);
+            actModel.APImodelGUID = applicationModelGuid;
+            actModel.ReturnValues = act.ActReturnValues;
+            actModel.MaxNumberOfRetries = act.MaxNumberOfRetries;
+            actModel.Active = true;
+            return actModel;
+        }
+
+        /// <summary>
+        /// This method is used to check if the app parameter is already added in the list if not then it will add the parameter
+        /// </summary>
+        /// <param name="aPIModel"></param>
+        /// <param name="act"></param>
+        /// <param name="actApiModel"></param>
+        private void AddAppParameters(ApplicationAPIModel aPIModel, Act act, ActWebAPIModel actApiModel)
+        {
+            try
+            {
+                ObservableList<AppModelParameter> lstParameters = null;
+                ObservableList<ActInputValue> actInputs = ((ActWebAPIBase)act).DynamicElements;
+                if (actInputs == null || actInputs.Count == 0)
+                {
+                    string requestBody = string.Empty;
+                    if (aPIModel.RequestBodyType == ApplicationAPIUtils.eRequestBodyType.TemplateFile)
+                    {
+                        int indexBF = aPIModel.ContainingFolder.IndexOf("BusinessFlows");
+                        string apiModelPath = aPIModel.ContainingFolder.Substring(0, (indexBF - 1));
+                        string fileUri = aPIModel.TemplateFileNameFileBrowser.Replace("~\\", apiModelPath);
+                        if (File.Exists(fileUri))
+                        {
+                            requestBody = System.IO.File.ReadAllText(fileUri);
+                        }
+                    }
+                    else
+                    {
+                        SetPropertyValue(aPIModel, nameof(ActWebAPIBase.Fields.RequestBody), nameof(ApplicationAPIModel.RequestBody), act);
+                        requestBody = Convert.ToString(aPIModel.RequestBody);
+                    }
+                    if (!string.IsNullOrEmpty(requestBody) && requestBody.StartsWith("{"))
```

Comment:
i guess we have netter way to check if body is XML or JSON, how we doing it when we parsing the web service action return respond body?